src/org/openvv/js/OVVAsset.js

```diff
@@ -540,6 +469,124 @@ OVVCheck.BEACON = 'beacon';
 */
 OVVCheck.GEOMETRY = 'geometry';
 
+function OVVBrowser(userAgent)
+{
+
+    var browserIDEnum = {
+        MSIE: 1,
+        Firefox: 2,
+        Chrome: 3,
+        Opera: 4,
+        safari: 5
+    };
+
+    /**
+     * Returns an object that contains the browser name, version, id and os if applicable
+     * @param {String} ua userAgent
+     */
+    function getBrowserDetailsByUserAgent(ua, t) {
+
+        var getData = function () {
+            var data = { ID: 0, name: '', version: '' };
+            var dataString = ua;
+            for (var i = 0; i < dataBrowsers.length; i++) {
+                // Fill Browser ID
+                if (dataString.match(new RegExp(dataBrowsers[i].brRegex)) != null) {
+                    data.ID = dataBrowsers[i].id;
+                    data.name = dataBrowsers[i].name;
+                    if (dataBrowsers[i].verRegex == null) {
+                        break;
+                    }
+                    //Fill Browser Version
+                    var brverRes = dataString.match(new RegExp(dataBrowsers[i].verRegex + '[0-9]*'));
```

Comment:
As a general note, I would prefer to have a class of the OS data (similar to the class we have for browser data).
Also, consider separating the logic of OS detection to a different method